smac/tae/execute_ta_run.py

```diff
@@ -159,17 +185,21 @@ def start(self, config:Configuration,
                                     "exit. The last incumbent can be found "
                                     "in the trajectory-file.")
 
-        # update SMAC stats
-        self.stats.ta_runs += 1
-        self.stats.ta_time_used += float(runtime)
-
         if self.run_obj == "runtime":
-            if status != StatusType.SUCCESS:
-                cost = cutoff * self.par_factor
-            else:
+            if cutoff is not None and runtime > self.par_factor * cutoff:
+                self.logger.warn("Returned running time is larger "
+                                 "than 2 times the passed cutoff time. "
+                                 "Clamping to {} x cutoff.".format(self.par_factor))
+                runtime = cutoff * self.par_factor
+            if status == StatusType.SUCCESS:
                 cost = runtime
+            else:
+                cost = cutoff * self.par_factor
```

Comment:
This statement is also true if cutoff is None. If this can happen, we don#t want to use cutoff\*par.
Is there a case where cutoff is None and we optimize runtime? If no, the first if can also be changed.